subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/DefaultFileLockManager.java

```diff
@@ -146,6 +145,9 @@ public DefaultFileLock(File target, LockOptions options, String displayName, Str
             LockStateSerializer stateProtocol = options.isUseCrossVersionImplementation() ? new Version1LockStateSerializer() : new DefaultLockStateSerializer();
             lockFileAccess = new LockFileAccess(lockFile, new LockStateAccess(stateProtocol));
             try {
+                if (whenContended != null) {
```

Comment:
⭕️ What does this change fix? I only see adjusted mock tests in this commit, but no integration test (maybe that comes later). Can you just point me to the test coverage?